### PR TITLE
Add memory

### LINKED ISSUE #64: SHRDLU Memory
Backend: 
- Keep history of messages 
- Keep history of enviroment
- Routes for sending history

Frontend:
- Fill everything in when loading

### PR DESCRIPTION
**Pull request by joshcwinton**
Adds basic memory to SHRDLU

Issue: Closes #64

### Frontend:
- Fetch messages from backend and update state when chat component mounts
### Backend:
- Keep a history of messages in the same format as `Chat.jsx` in a global variable in `environment.py`
- Keep a list of environment grids in a global variable in `environment.py`
- Create functions for accessing history data
- Create a route for fetching message history
- Create a route for accessing environment history

### CODE REVIEW

File: backend/app.py [hunk 1]
```diff
@@ -73,6 +74,20 @@ def environment_image():
...
 
 
+@app.route('/messages', methods=['GET'])
+def messages():
+    if request.method == 'GET':
+        return jsonify({"messages": getMessages()})
+    return None
+
+
+@app.route('/history', methods=['GET'])
```
Comment (JasperCheung) [inline backend/app.py:84]: is this being used anywhere yet?

Comment (joshcwinton) [reply backend/app.py:84]: No not yet

**Review by svhnstorage — APPROVED** [review]
LGTM but the front end is more for Jasper

**Review by JasperCheung — APPROVED** [review]
lgtm

**Pull request merged**